### PR TITLE
[Flang] Support -mrvv-vector-bits flag

### PR DESCRIPTION
**Pull request by lukel97**
This patch adds support for the -mrvv-vector-bits flag in the Flang driver, and
translates them to -mvscale-min/-mvscale-max.

The code was copied from the Clang toolchain (similarly to what was done for
AArch64's -msve-vector-bits flag) so it also supports the same
-mrvv-vector-bits=zvl mode.

Note that Flang doesn't yet define the __riscv_v_fixed_vlen macro, so the help
text has been updated to highlight that it's only defined for Clang.

### CODE REVIEW

File: clang/include/clang/Driver/Options.td
Comment (banach-space) [inline clang/include/clang/Driver/Options.td:4594]: I am very against using `HelpText` like this - it makes compiler "help" output (i.e. `flang-new -help`) unusable. Please split into `HelpText` and `DocBrief` and keep `HelpText` short and concise.

**Review by banach-space — COMMENTED** [review]
Makes sense to me, but somebody actually familiar with RISC-V should review as well.

**Comment by banach-space** [comment]
Thanks for addressing my comment! 

The overall logic LGTM, but please wait for somebody to review the finer RISC-V details before landing this.

**Review by preames — APPROVED** [review]
LGTM

We should explore options for merging the option processing code for options supported by both clang and flang, but that's explicitly future work.

**Pull request merged**
